tests/components/husqvarna_automower/test_switch.py

```diff
@@ -216,29 +217,49 @@ async def test_work_area_switch_commands(
     assert len(mocked_method.mock_calls) == 2
 
 
-async def test_zones_deleted(
+async def test_add_stay_out_zone(
     hass: HomeAssistant,
     mock_automower_client: AsyncMock,
     mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
     entity_registry: er.EntityRegistry,
 ) -> None:
-    """Test if stay-out-zone is deleted after removed."""
-
+    """Test adding a stay out zone in runtime."""
+    await setup_integration(hass, mock_config_entry)
+    entry = hass.config_entries.async_entries(DOMAIN)[0]
+    current_entites = len(
+        er.async_entries_for_config_entry(entity_registry, entry.entry_id)
+    )
     values = mower_list_to_dictionary_dataclass(
         load_json_value_fixture("mower.json", DOMAIN)
     )
-    await setup_integration(hass, mock_config_entry)
-    current_entries = len(
-        er.async_entries_for_config_entry(entity_registry, mock_config_entry.entry_id)
-    )
 
-    del values[TEST_MOWER_ID].stay_out_zones.zones[TEST_ZONE_ID]
+    values[TEST_MOWER_ID].stay_out_zones.zones.update(
+        {
+            TEST_VARIABLE_ZONE_ID: Zone(
+                name="future_zone",
+                enabled=True,
+            )
+        }
+    )
     mock_automower_client.get_status.return_value = values
-    await hass.config_entries.async_reload(mock_config_entry.entry_id)
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
     await hass.async_block_till_done()
-    assert len(
-        er.async_entries_for_config_entry(entity_registry, mock_config_entry.entry_id)
-    ) == (current_entries - 1)
+    current_entites_after_addition = len(
+        er.async_entries_for_config_entry(entity_registry, entry.entry_id)
+    )
+    assert current_entites_after_addition == current_entites + 1
+    values[TEST_MOWER_ID].stay_out_zones.zones.pop(TEST_VARIABLE_ZONE_ID)
+    # values[TEST_MOWER_ID].stay_out_zones.zones.pop(TEST_ZONE_ID)
```

Comment:
When uncommenting this line there is the following error:
```
 tests/components/husqvarna_automower/test_switch.py ⨯                                                                                                                                                  100% ██████████--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/logging/__init__.py", line 1163, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1870, in call_exception_handler
    ctx.run(self._exception_handler, self, context)
  File "/workspaces/core/tests/conftest.py", line 578, in exc_handle
    orig_exception_handler(loop, context)
  File "/workspaces/core/homeassistant/runner.py", line 138, in _async_loop_exception_handler
    logger.error(
Message: 'Error doing job: %s (%s): %s'
```
I can't absolutley not explain why removing this zone doesn't work, but removing the before created variable zone can be removed.

Reply:
Interesting. Can you maybe try replacing values with `mock_automower_client.get_status.return_value`?